Content.Server/_NF/Shipyard/Systems/ShipyardSystem.cs

```diff
@@ -279,6 +287,49 @@ public ShipyardSaleResult TrySellShuttle(EntityUid stationUid, EntityUid shuttle
         return result;
     }
 
+    private void CleanGrid(EntityUid grid, EntityUid destination)
+    {
+        var xform = Transform(grid);
+        var enumerator = xform.ChildEnumerator;
+        var entitiesToPreserve = new List<EntityUid>();
+
+        while (enumerator.MoveNext(out var child))
+        {
+            HasPreserveOnSaleComp(child, ref entitiesToPreserve);
+        }
+        foreach (var ent in entitiesToPreserve)
+        {
+            // Teleport this item and all its children to the floor (or space).
+            _transform.SetCoordinates(ent, new EntityCoordinates(destination, 0, 0));
+            _transform.AttachToGridOrMap(ent);
+        }
+    }
+
+    // checks if something has the ShipyardPreserveOnSaleComponent and if it does, adds it to the list
+    private void HasPreserveOnSaleComp(EntityUid entity, ref List<EntityUid> output)
```

Comment:
```suggestion
    private void FindEntitiesToPreserve(EntityUid entity, ref List<EntityUid> output)
```
HasX should return a bool, you're doing things.